packages/preview/machiatto/0.2.0/lib.typ

```diff
@@ -0,0 +1,332 @@
+
+      let before = query(target.before(here()))
+      if before.len() > 0 {
+        let current = before.last()
+        let gap = 1.75em
+        let chapter = upper(text(size: 0.68em, current.body))
+        if current.numbering != none {
+          if is-odd {
+            align(aln)[#chapter #h(gap) #i]
+          } else {
+            align(aln)[#i #h(gap) #chapter]
+          }
+        }
+      }
+    },
+  )
+
+  // Configure equation numbering.
+
+  set math.equation(numbering: "(1)")
+
+  {
+    // Configure heading numbering.
+    set heading(numbering: "1.")
+
+    // Start chapters on a new page.
+
+    show heading.where(level: 1): it => {
+      if chapter-pagebreak {
+        pagebreak(weak: true)
+      }
+      it
+    }
+    body
+  }
+
+  if bibliography != none {
+    pagebreak()
+    show std-bibliography: set text(0.85em)
+    // Use default paragraph properties for bibliography.
+
+    show std-bibliography: set par(leading: 0.65em, justify: false, linebreaks: auto)
+    bibliography
+  }
+}
+
+#let def(title: none, fill: color.rgb("E0F2F1"), body) = {
+  block(
+    width: 100%,
+    inset: 10pt,
+    fill: fill,
+    radius: 5pt,
+    stroke: 1pt,
+    [
+      #text(weight: "bold")[#underline(stroke: 1pt)[Definition: #title]]
+
+      #body
+    ],
+  )
+}
+
+#let todo(body) = {
+  block(
+    width: 100%,
+    inset: 10pt,
+    fill: yellow,
+    radius: 4pt,
+    [
+      #text(weight: "bold")[TODO]
+
+      #body
+    ],
+  )
+}
+
+#let note(body) = {
+  block(
+    width: 100%,
+    inset: 10pt,
+    fill: color.rgb("FFECB3"),
+    radius: 4pt,
+    stroke: 1pt,
+    [
+      #text(weight: "bold")[Note:]
+
+      #body
+    ],
+  )
+}
+
+#let frame(title: none, body, fill: none, line-numbers: true, line-number-color: gray) = {
+  let stroke = black + 1pt
+  let radius = 5pt
+
+  box(stroke: stroke, radius: radius)[
+    #if title != none {
+      block(
+        stroke: stroke,
+        fill: fill,
+        inset: 0.5em,
+        width: 100%,
+        below: 0em,
+        radius: (top-left: radius, top-right: radius),
+        align(center)[*#title*],
+      )
+    }
+    #block(
+      width: 100%,
+      inset: (x: 0em, y: 0.5em),
+      {
+        if line-numbers {
+          // Split the body into lines
+          let lines = body.text.split("\n")
+          // Create a table with line numbers and code
+
+          table(
+            columns: (auto, 1fr),
+            inset: (x: 0.4em, y: 0.3em),
+            stroke: (y: 0.5pt + rgb(220, 220, 220)),
+            ..lines.enumerate().map(((i, line)) => {
+              (
+                align(right)[#text(fill: line-number-color)[#(i + 1)]],
+                align(left)[#raw(lang: body.lang, line)]
+              )
+            }).flatten(),
+          )
+        } else {
+          body
+        }
+      },
+    )
+  ]
+}
+
+#let code-file(title: str, file-contents: str, lang, fill, line-numbers: true, line-number-color: gray) = {
```

Comment:
If you want to pass the file content as an unnamed argument (making it positional), as you seem to by looking at how you call this function in your template, you should not provide it with a default value. It is then the order in which they will appear in the function call that will determine which value correspond to which argument.

By the way, in Typst, the value coming after `:` in arguments is never a type but a default value, and I don't think having `str` as a default value makes sense, you probably want `""` or `none` instead.